client/my-sites/domains/domain-management/dns/dns-menu-options-button.tsx

```diff
@@ -3,49 +3,56 @@ import { Button } from '@automattic/components';
 import { Icon, moreVertical, redo } from '@wordpress/icons';
 import { useI18n } from '@wordpress/react-i18n';
 import { useCallback, useRef, useState } from 'react';
+import { connect } from 'react-redux';
 import PopoverMenu from 'calypso/components/popover-menu';
 import PopoverMenuItem from 'calypso/components/popover-menu/item';
-import wpcom from 'calypso/lib/wp';
-import './dns-breadcrumb-button.scss';
+import { updateDns } from 'calypso/state/domains/dns/actions';
+import { errorNotice, successNotice } from 'calypso/state/notices/actions';
 import RestoreDefaultARecordsDialog from './restore-default-a-records-dialog';
 import { DnsMenuOptionsButtonProps, RestoreDialogResult } from './types';
 
 function DnsMenuOptionsButton( {
 	domain,
 	pointsToWpcom,
-	onSuccess,
-	onError,
+	dns,
+	dispatchUpdateDns,
+	dispatchSuccessNotice,
+	dispatchErrorNotice,
 }: DnsMenuOptionsButtonProps ): JSX.Element {
 	const { __ } = useI18n();
 
 	const [ isMenuVisible, setMenuVisible ] = useState( false );
 	const [ isRestoreDialogVisible, setRestoreDialogVisible ] = useState( false );
 	const optionsButtonRef = useRef( null );
-	const restoreRecordsErrorMessage = __(
-		'An unexpected error occurred when trying to restore your DNS records. Please try again later.'
-	);
 
 	const toggleMenu = useCallback( () => {
 		setMenuVisible( ! isMenuVisible );
 	}, [ isMenuVisible ] );
 
 	const closeMenu = useCallback( () => setMenuVisible( false ), [] );
 
-	const restoreDefaultRecords = useCallback( async () => {
-		const wpcomDomain = wpcom.domain( domain );
+	const getRecordsToRemove = useCallback( () => {
+		const dnsRecords = dns?.records ?? [];
```

Comment:
Minor:
Could the `dns` prop be falsy? 
If so, then I think we should make it optional on the `DnsMenuOptionsButtonProps` type.
If not, then we wouldn't need this optional check (`dns?.records`).

Reply:
Good call! I'll remove the check.